### ISSUE TITLE
[Feature Request] Upgrade google-cloud-storage to v2

### ISSUE THREAD

**Issue by cwperks** [opensearch-project/OpenSearch#16156]
### Is your feature request related to a problem? Please describe

Coming from conversation on https://github.com/opensearch-project/OpenSearch/pull/16072

The [repository-gcs plugin](https://github.com/opensearch-project/OpenSearch/tree/main/plugins/repository-gcs) currently depends on v1 of the google-cloud-storage dependencies. Dependabot has been creating automatic updates which consistently fail precommit and tests and cannot be merged without further modifications.

I have tried to upgrade this dependency, but consistently receive errors on `GoogleCloudStorageBlobContainerRetriesTests.testWriteLargeBlob` post-upgrade which uses the resumable upload feature of google cloud storage.

The test can be run using `./gradlew :plugins:repository-gcs:test --tests GoogleCloudStorageBlobContainerRetriesTests.testWriteLargeBlob -i`. When running that test, I have to add an `image:` tag to `test/fixtures/gcs-fixture/docker-compose.yml` in all services in order to bring up the containers successfully. 

### Describe the solution you'd like

Update to v2 for `com.google.cloud:google-cloud-storage` dependency.

### Related component

Plugins

### Additional context

There were lines added to dependabot.yml to prevent dependabot from making PRs on a major version update for this dependency. After upgrading to v2, make sure to remove [these lines](https://github.com/opensearch-project/OpenSearch/blob/main/.github/dependabot.yml#L726-L728) from dependabot.yml.

**Comment by Yashovardhan08** [comment]
@cwperks @sandeshkr419, I am interested, looking if someone could help me get started and guide me if I get stuck.

**Comment by cwperks** [comment]
@Yashovardhan08 Certainly, when I tried to make an update it didn't seem like a straightforward change, but I didn't have time to dig deeper into the tests within the repository-gcs plugin and any changes to resumable upload.

There's another outdated dependency that may be more straightforward which I would suggest starting with. In the discovery-gce plugin [google-api-client is currently using v1](https://github.com/opensearch-project/OpenSearch/blob/main/plugins/discovery-gce/build.gradle#L22) and can be upgraded to the[ latest v2 version of the dependency](https://mvnrepository.com/artifact/com.google.api-client/google-api-client). I would recommend starting with the update in discovery-gce plugin and then moving to this issue which is known to require test updates.

**Comment by dblock** [comment]
[Catch All Triage - [1](https://github.com/dblock), [2](https://github.com/reta)]

**Comment by Yashovardhan08** [comment]
Please review the PR [here](https://github.com/opensearch-project/OpenSearch/pull/16460) for com.google.api-client upgrade from v1 to v2.
